WordPress/gutenberg
# Freeform: Move modal styles to the correct file

## What?
This is a follow-up to #50384.

PR moves block editor-specific styles from `style.scss` to `editor.scss`.

## Why?
* The editor-only styles should live in `editor.scss`.
* The `packages/block-library/src/style.scss` is bundled and enqueued on the front end; there's no need to have this CSS when viewing the site.

## Testing Instructions
1. Open a Post or Page. 
2. Insert the Classic block.
3. Click on the "Edit" toolbar button.
4. Confirm modal styles are applied as before.

## Screenshots or screencast 

![CleanShot 2023-05-23 at 10 52 38](https://github.com/WordPress/gutenberg/assets/240569/babc2429-b32c-419d-a433-17cbafeedc31)
